src/utils/check-pr/github_client.py

```diff
@@ -0,0 +1,200 @@
+import argparse
+import os
+import sys
+import time
+from pathlib import Path
+from typing import Callable, Optional, Tuple, TypeVar
+
+import requests
+from github import Github
+
+A = TypeVar("A")
+
+
+def wait(func: Callable[[], Tuple[bool, str, A]], frequency: float = 1.0) -> A:
+    """
+    Wait until the provided func returns True.
+
+    Provides user feedback via a spinner if stdout is a TTY.
+    """
+
+    isatty = sys.stdout.isatty()
+    frames = ["-", "\\", "|", "/"]
+    waited = False
+    last_message = None
+    result = None
+
+    try:
+        while True:
+            result = func()
+            if result[0]:
+                break
+            message = result[1]
+
+            if isatty:
+                if last_message:
+                    if last_message == message:
+                        sys.stdout.write("\b" * (len(last_message) + 2))
+                    else:
+                        sys.stdout.write("\n")
+                sys.stdout.write("%s %s" % (frames[0], message))
+                sys.stdout.flush()
+            elif last_message != message:
+                print(message, flush=True)
+
+            last_message = message
+            waited = True
+            time.sleep(frequency)
+            frames.sort(key=frames[0].__eq__)
+    finally:
+        if waited and isatty:
+            print(flush=True)
+
+    return result[2]
+
+
+class GithubClient:
```

Comment:
This is just the Downloaded class that existed in check-pr.py
I just renamed it to `GithubClient` without changing the logic

Reply:
Is there any new code here? Or is it all pulled from the original check-pr

Reply:
no new code.